### PR TITLE
Store interface impl

### PR DESCRIPTION
**Pull request by tianliplus**
## Related issue number

Fixes

### CODE REVIEW

File: interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs [hunk 1]
```diff
@@ -18,38 +18,63 @@ use crate::v2::api::types::{Vertex, Edge, EdgeRelation};
...
         edge_relation: Option<&EdgeRelation>,
         condition: Option<&Condition>,
         property_ids: Option<&Vec<PropertyId>>,
     ) -> Result<Records<Self::E>>;
 
+    /// Returns out edges of vertex `vertex_id`, filtered by `label_id` and
+    /// `condition` optionally.
+    ///
+    /// Properties of the edges are filtered by the `property_ids` optionally.
     fn get_out_edges(
```
Comment (LiSu) [inline interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs:78]: Should get_out_edge have label_id and condition as its input params?

File: interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs [hunk 2]
```diff
@@ -58,6 +83,10 @@ pub trait PartitionSnapshot: Send + Sync {
         property_ids: Option<&Vec<PropertyId>>,
     ) -> Result<Records<Self::E>>;
 
+    /// Returns in edges of vertex `vertex_id`, filtered by `label_id` and
+    /// `condition` optionally.
+    ///
+    /// Properties of the edges are filtered by the `property_ids` optionally.
     fn get_in_edges(
```
Comment (LiSu) [inline interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs:90]: The same problem as get_out_edge.

File: interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs [hunk 3]
```diff
@@ -66,18 +95,23 @@ pub trait PartitionSnapshot: Send + Sync {
...
     fn get_in_degree(
         &self,
         vertex_id: VertexId,
         edge_relation: &EdgeRelation,
     ) -> Result<usize>;
 
+    /// Returns the `k`th out edge of vertex `vertex_id` in `edge_relation`.
+    ///
+    /// Properties of the edge are filtered by the `property_ids` optionally.
     fn get_kth_out_edge(
```
Comment (LiSu) [inline interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs:115]: Should get_kth_out_edge have edge_relation as one of its input pram?

File: interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs [hunk 4]
```diff
@@ -86,6 +120,9 @@ pub trait PartitionSnapshot: Send + Sync {
         property_ids: Option<&Vec<PropertyId>>,
     ) -> Result<Option<Self::E>>;
 
+    /// Returns the `k`th in edge of vertex `vertex_id` in `edge_relation`.
+    ///
+    /// Properties of the edge are filtered by the `property_ids` optionally.
     fn get_kth_in_edge(
```
Comment (LiSu) [inline interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs:126]: The same problem as get_kth_out_edge

File: interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs (hunk 1 above)
Comment (tianliplus) [reply interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs:78]: line 81~82

File: interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs (hunk 2 above)
Comment (tianliplus) [reply interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs:90]: line 93~94

File: interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs (hunk 3 above)
Comment (tianliplus) [reply interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs:115]: line 118

File: interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs (hunk 4 above)
Comment (tianliplus) [reply interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs:126]: line 129

**Review by LiSu — APPROVED** [review]
This implementation aims to facilitate an end-to-end C++ to Rust test, and needs further optimization with regard to the performance and API specifications.

**Pull request merged**